cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,305 @@
+		let free_bal_after_reap = RococoBalances::free_balance(RococoRelaySender::get());
+
+		// free balance should be greater than before reap
+		assert!(free_bal_after_reap > free_bal_before_reap);
+		assert_eq!(free_bal_after_reap, free_bal_before_reap + total_deposit);
+	});
+}
+
+fn assert_reap_parachain(id: Identity) {
+	// 6. assert on Parachain
+	PeopleRococo::execute_with(|| {
+		type RuntimeEvent = <PeopleRococo as Chain>::RuntimeEvent;
+		let reserved_bal = PeopleRococoBalances::reserved_balance(PeopleRococoSender::get());
+		let id_deposit = id_deposit_parachain(&id.parachain);
+		let subs_deposit = SubAccountDepositParachain::get();
+		let total_deposit = subs_deposit + id_deposit;
+
+		assert_expected_events!(
+			PeopleRococo,
+			vec![
+				RuntimeEvent::Balances(BalancesEvent::Deposit { .. }) => {},
+				RuntimeEvent::Balances(BalancesEvent::Endowed { .. }) => {},
+				RuntimeEvent::Balances(BalancesEvent::Reserved { who, amount }) => {
+					who: *who == PeopleRococoSender::get(),
+					amount: *amount == id_deposit,
+				},
+				RuntimeEvent::Balances(BalancesEvent::Reserved { who, amount }) => {
+					who: *who == PeopleRococoSender::get(),
+					amount: *amount == subs_deposit,
+				},
+				RuntimeEvent::IdentityMigrator(
+					polkadot_runtime_common::identity_migrator::Event::DepositUpdated {
+						who, identity, subs
+					}) => {
+					who: *who == PeopleRococoSender::get(),
+					identity: *identity == id_deposit,
+					subs: *subs == subs_deposit,
+				},
+				RuntimeEvent::MessageQueue(pallet_message_queue::Event::Processed { ..}) => {},
+			]
+		);
+
+		// reserved balance should be equal to total deposit calculated on the Parachain
+		assert_eq!(reserved_bal, total_deposit);
+
+		// Should have at least one ED after in free balance after the reap.
+		assert!(PeopleRococoBalances::free_balance(PeopleRococoSender::get()) >= PEOPLE_ROCOCO_ED);
+	});
+}
+
+#[test]
+fn on_reap_identity_works_for_first_instance() {
+	let ids = identities();
+	let total_deposit = set_id_relay(ids[0].clone());
+	assert_set_id_parachain(ids[0].clone());
+	assert_reap_id_relay(total_deposit);
+	assert_reap_parachain(ids[0].clone());
+}
+
+#[test]
+fn on_reap_identity_works_for_second_instance() {
+	let ids = identities();
+	let total_deposit = set_id_relay(ids[1].clone());
+	assert_set_id_parachain(ids[1].clone());
+	assert_reap_id_relay(total_deposit);
+	assert_reap_parachain(ids[1].clone());
+}
+
+#[test]
+fn on_reap_identity_works_for_third_instance() {
+	let ids = identities();
+	let total_deposit = set_id_relay(ids[2].clone());
+	assert_set_id_parachain(ids[2].clone());
+	assert_reap_id_relay(total_deposit);
+	assert_reap_parachain(ids[2].clone());
+}
+
+#[test]
+fn on_reap_identity_works_for_fourth_instance() {
+	let ids = identities();
+	let total_deposit = set_id_relay(ids[3].clone());
+	assert_set_id_parachain(ids[3].clone());
+	assert_reap_id_relay(total_deposit);
+	assert_reap_parachain(ids[3].clone());
+}
```

Comment:
Does this not work?
```suggestion
#[test]
fn on_reap_identity_works() {
	let ids = identities();
	for id in ids {
		let total_deposit = set_id_relay(id.clone());
		assert_set_id_parachain(id.clone());
		assert_reap_id_relay(total_deposit);
		assert_reap_parachain(id);
	}
}
```

Reply:
I tried this initially, but on the second and subsequent times we loop `assert_set_id_parachain` fails because `assert_eq!(reserved_bal + free_bal, 0);` is not true. `free_bal` on `PeopleRococo` account has balance after the first reap. We need to start at genesis state each time. 

If there was some way to set the block number back to 0 (go back to genesis) then we could loop. But I found no function to do this and it's probably outside the scope of this PR to make one.

Reply:
I don't see how setting the block number to 0 is the only way to achieve this.
For example you could set a variable to 0, pass it to the assert_set_id_parachain function and use that as the comparison in the assertion. Before the next loop update it to whatever is left on the chain from the previous case and it solves this problem without duplication of code.